adapters/nextmillennium/nextmillennium.go

```diff
@@ -99,12 +99,28 @@ func (adapter *adapter) buildAdapterRequest(prebidBidRequest *openrtb2.BidReques
 }
 
 func createBidRequest(prebidBidRequest *openrtb2.BidRequest, params *openrtb_ext.ImpExtNextMillennium) *NextMillenniumBidRequest {
+	placementID := params.PlacementID
 	bidRequest := NextMillenniumBidRequest{
 		ID:   prebidBidRequest.ID,
 		Test: uint8(prebidBidRequest.Test),
 	}
-	bidRequest.Ext.Prebid.StoredRequest.ID = params.PlacementID
 
+	if params.GroupID != nil {
+		domain := ""
+		size := ""
+		if prebidBidRequest.Site != nil && prebidBidRequest.Site.Domain != "" {
+			domain = prebidBidRequest.Site.Domain
+		}
+		if prebidBidRequest.App != nil && prebidBidRequest.App.Domain != "" {
+			domain = prebidBidRequest.App.Domain
+		}
+		if prebidBidRequest.Imp[0].Banner != nil {
+			size = fmt.Sprintf("%dx%d", *prebidBidRequest.Imp[0].Banner.W, *prebidBidRequest.Imp[0].Banner.H)
```

Comment:
You will get a panic if `.W` or `.H` are not defined. As far as I'm aware, there's no guarantee from the core logic that those will exist. Please add a json test case where the banner is defined with W and H missing and update the logic to check those are not nil.

Reply:
Done!